src/v0/sources/shopify/util.js

```diff
@@ -113,98 +131,73 @@ const getRudderIdFromNoteAtrributes = (noteAttributes, field) => {
- */
-const getAnonymousIdFromDb = async (cartToken, message, metricMetadata) => {
-  let anonymousId;
-  stats.increment('shopify_redis_calls', {
-    type: 'get',
-    ...metricMetadata,
-  });
-  let failCount = 0;
-  try {
-    anonymousId = await RedisDB.getVal(`${cartToken}`, 'anonymousId');
-  } catch (e) {
-    failCount += 1;
-  }
-  stats.counter('shopify_redis_failures', failCount, {
-    type: 'get',
-    ...metricMetadata,
-  });
-  if (isDefinedAndNotNull(anonymousId)) {
-    return anonymousId;
-  }
-  stats.increment('shopify_redis_no_val', {
-    ...metricMetadata,
-    event: message.event,
-  });
-  // If there is no mapping or there is some error in connection we return null
-  return null;
-};
-
-/**
- * This function retrieves anonymousId in folowing steps:
- * 1. Checks for `rudderAnonymousId` in `note_atrributes`
+ * This function retrieves anonymousId and sessionId in folowing steps:
+ * 1. Checks for `rudderAnonymousId`and `rudderSessionId in `note_atrributes`
  * 2. if redis is enabled checks in redis
  * 3. This means we don't have `anonymousId` and hence events CAN NOT be stitched and we check for cartToken
  *    a. if cartToken is available we return its hash value
  *    b. else we check if the event is an SHOPIFY_ADMIN_ONLY_EVENT
  *       -> if true we return `null`;
  *       -> else we don't have any identifer (very edge case) we return `random anonymousId`
- * @param {*} message
- * @param {*} metricMetadata
- * @returns
+ *    No Random SessionId is generated as its not a required field
+ * @param {*} message 
+ * @param {*} metricMetadata 
+ * @returns 
  */
-const getAnonymousId = async (message, metricMetadata) => {
+const getAnonymousIdAndSessionId = async (message, metricMetadata, redisData = null) => {
   let anonymousId;
+  let sessionId;
   const noteAttributes = message.properties?.note_attributes;
-  // Giving Priority to note_attributes to fetch rudderAnonymousId over Redis ue to better functionality
+  // Giving Priority to note_attributes to fetch rudderAnonymousId over Redis due to better efficiency
   if (isDefinedAndNotNull(noteAttributes)) {
-    anonymousId = getRudderIdFromNoteAtrributes(noteAttributes, 'rudderAnonymousId');
+    anonymousId = getRudderIdFromNoteAtrributes(noteAttributes, "rudderAnonymousId");
+    sessionId = getRudderIdFromNoteAtrributes(noteAttributes, "rudderSessionId");
   }
-  // falling back to cartToken mapping or its hash in case no rudderAnonymousId is found
-  if (isDefinedAndNotNull(anonymousId)) {
-    return anonymousId;
+  // falling back to cartToken mapping or its hash in case no rudderAnonymousId or rudderSessionId is found
+  if (isDefinedAndNotNull(anonymousId) && isDefinedAndNotNull(sessionId)) {
+    return { anonymousId, sessionId };
   }
   const cartToken = getCartToken(message);
   if (!isDefinedAndNotNull(cartToken)) {
     if (SHOPIFY_ADMIN_ONLY_EVENTS.includes(message.event)) {
-      return null;
+      return { anonymousId, sessionId };
```

Comment:
`anonymousId, sessionId` could be undefined here

Reply:
Yeah it can be and we are handling these added test cases as well for these